parsers/CA_AB.py

```diff
@@ -1,161 +1,153 @@
-            'solar': float(maximum_capability['SOLAR']),
-            'wind': float(maximum_capability['WIND']),
-            'biomass': float(maximum_capability['OTHER']),
-            'unknown': float(maximum_capability['DUAL FUEL']),
-            'coal': float(maximum_capability['COAL'])
-        },
-        'source': 'ets.aeso.ca',
+        'datetime': get_csd_report_timestamp(response.text),
+        'sortedZoneKeys': sorted_zone_keys,
+        'netFlow': float(flows[sorted_zone_keys]),
+        'source': URL.netloc,
     }
 
 
-def fetch_price(zone_key='CA-AB', session=None, target_datetime=None, logger=None) -> dict:
-    """Requests the last known power price of a given country."""
+def fetch_price(zone_key=DEFAULT_ZONE_KEY,
+                session=None,
+                target_datetime=None,
+                logger=None) -> dict:
+    """Request the last known power price of a given country."""
     if target_datetime:
-        raise NotImplementedError('This parser is not yet able to parse past dates')
-
-    r = session or requests.session()
-    url = 'http://ets.aeso.ca/ets_web/ip/Market/Reports/SMPriceReportServlet?contentType=html/'
-    response = r.get(url)
-
-    df_prices = pd.read_html(response.text, match='Price', index_col=0, header=0)
-    prices = df_prices[1]
-
-    data = {}
-
-    for rowIndex, row in prices.iterrows():
-        price = row['Price ($)']
-        if (isfloat(price)):
-            hour = int(rowIndex.split(' ')[1]) - 1
-            data[rowIndex] = {
-                'datetime': arrow.get(rowIndex, 'MM/DD/YYYY').replace(hour=hour, tzinfo=ab_timezone).datetime,
-                'zoneKey': zone_key,
-                'currency': 'CAD',
-                'source': 'ets.aeso.ca',
-                'price': float(price),
-            }
-
-    return [data[k] for k in sorted(data.keys())]
-
-
-def fetch_exchange(zone_key1='CA-AB', zone_key2='CA-BC', session=None, target_datetime=None, logger=None) -> dict:
-    """Requests the last known power exchange (in MW) between two countries."""
+        raise NotImplementedError('Currently unable to scrape historical data')
+    session = session or requests.Session()
+    response = session.get(f'{URL_STRING}/SMPriceReportServlet',
+                           params={'contentType': 'csv'})
+    return [
+        {
+            'currency': 'CAD',
+            'datetime': arrow.get(row[0], 'MM/DD/YYYY HH', tzinfo=TIMEZONE)
+                             .datetime,
+            'price': float(row[1]),
+            'source': URL.netloc,
+            'zoneKey': zone_key,
+        } for row in csv.reader(response.text
+                                        .split('\r\n\r\n')[2]
+                                        .splitlines()[1:]) if row[1] != '-']
+
+
+def fetch_production(zone_key=DEFAULT_ZONE_KEY,
+                     session=None,
+                     target_datetime=None,
+                     logger=None) -> dict:
```

Comment:
`validation.validate` expects a valid `logger` argument, but it defaults to `None` here. I should change this to `logger=logging.getLogger(__name__)`.